org.lflang/src/org/lflang/generator/ts/TSConstructorGenerator.kt

```diff
@@ -67,61 +66,62 @@ class TSConstructorGenerator (
+//            }
+//            return """
+//            super(federationID, ${federate.id}, ${port},
+//                "${federationRTIProperties()["host"]}",
+//                timeout, keepAlive, fast, success, fail);
+//            """
+//        }
+        else {
             return "super(parent);"
         }
     }
 
     // If the app is federated, register its
     // networkMessageActions with the RTIClient
-    private fun generateFederatePortActionRegistrations(networkMessageActions: List<Action>): String {
+    private fun generateFederatePortActionRegistrations(): String {
         var fedPortID = 0;
         val connectionInstantiations = LinkedList<String>()
-        for (nAction in networkMessageActions) {
-            val registration = """
-                this.registerFederatePortAction(${fedPortID}, this.${nAction.name});
-                """
-            connectionInstantiations.add(registration)
-            fedPortID++
-        }
+        // FIXME: Move to FedGenerator
+//        for (nAction in networkMessageActions) {
+//            val registration = """
+//                this.registerFederatePortAction(${fedPortID}, this.${nAction.name});
+//                """
+//            connectionInstantiations.add(registration)
+//            fedPortID++
+//        }
         return connectionInstantiations.joinToString("\n")
     }
 
     // Generate code for registering Fed IDs that are connected to
     // this federate via ports in the TypeScript's FederatedApp.
     // These Fed IDs are used to let the RTI know about the connections
     // between federates during the initialization with the RTI.
-    fun generateFederateConfigurations(): String {
-        val federateConfigurations = LinkedList<String>()
-        if (reactor.isFederated) {
-            for ((key, _) in federate.dependsOn) {
-                // FIXME: Get delay properly considering the unit instead of hardcoded BigInt(0).
-                federateConfigurations.add("this.addUpstreamFederate(${key.id}, BigInt(0));")
-            }
-            for ((key, _) in federate.sendsTo) {
-                federateConfigurations.add("this.addDownstreamFederate(${key.id});")
-            }
-        }
-        return federateConfigurations.joinToString("\n")
-    }
+    // FIXME: Move this to FedGenerator
+//    fun generateFederateConfigurations(): String {
+//        val federateConfigurations = LinkedList<String>()
+//        if (reactor.isFederated) {
+//            for ((key, _) in federate.dependsOn) {
+//                // FIXME: Get delay properly considering the unit instead of hardcoded BigInt(0).
+//                federateConfigurations.add("this.addUpstreamFederate(${key.id}, BigInt(0));")
+//            }
+//            for ((key, _) in federate.sendsTo) {
+//                federateConfigurations.add("this.addDownstreamFederate(${key.id});")
+//            }
+//        }
+//        return federateConfigurations.joinToString("\n")
+//    }
```

Comment:
General question for the commented code blocks "FIXME: Move this to FedGenerator":

Is TS generator going to somehow inherit the FedGenerator class? How can target-specific codes be included in the FedGenerator?

Reply:
There is an extension for TS [here](https://github.com/lf-lang/lingua-franca/blob/fed-gen/org.lflang/src/org/lflang/federated/extensions/TSExtension.java). This extension is used by the FedGenerator to produce target-specific code in the generated .lf file for each federate.